Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,177 @@
+//  ReceiveViewController.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import Foundation
+import UIKit
+import Dcrlibwallet
+
+class ReceiveViewController: UIViewController, UIDocumentInteractionControllerDelegate {
+    @IBOutlet weak var moreMenuButton: UIButton!
+    @IBOutlet weak var mainContentViewHolder: RoundedView!
+    @IBOutlet weak var selectedAccountView: UIView!
+    @IBOutlet weak var accountNameLabel: UILabel!
+    @IBOutlet weak var walletNameLabel: UILabel!
+    @IBOutlet weak var totalAccountBalanceLabel: UILabel!
+    @IBOutlet private var addressQRCodeImageView: UIImageView!
+    @IBOutlet weak var walletAddressLabel: UILabel!
+    @IBOutlet weak var syncInProgressLabel: UILabel!
+
+    var tapGesture = UITapGestureRecognizer()
+    var oldAddress = ""
+
+    var selectedWallet: DcrlibwalletWallet?
+    var selectedAccount: DcrlibwalletAccount?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.setupUI()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.checkSyncStatus()
+    }
+
+    func setupUI() {
+        self.addressQRCodeImageView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.walletAddressLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.selectedAccountView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.showAccountSelectorDialog)))
+    }
+
+    @objc func copyAddress() {
+        DispatchQueue.main.async {
+            UIPasteboard.general.string = self.walletAddressLabel.text!
+            Utils.showBanner(parentVC: self, type: .success, text: LocalizedStrings.walletAddrCopied)
+        }
+    }
+
+    @objc func showAccountSelectorDialog(_ sender: Any) {
+        AccountSelectorDialog.show(sender: self,
+                                 title: LocalizedStrings.receivingAccount,
+                                 selectedWallet: selectedWallet,
+                                 selectedAccount: self.selectedAccount,
+                                 callback: self.updateSelectedAccount)
+    }
+
+    private func checkSyncStatus() {
+        let accountsFilterFn: (DcrlibwalletAccount) -> Bool = { $0.totalBalance > 0 || $0.name != "imported" }
+        guard let wallet = WalletLoader.shared.wallets.map({ Wallet.init($0, accountsFilterFn: accountsFilterFn) }).first,
+            let account = wallet.accounts.first else {
+                self.moreMenuButton.isEnabled = false
+                self.mainContentViewHolder.isHidden = true
+                self.syncInProgressLabel.isHidden = false
```

Comment:
This label should only be visible if the selected wallet was restored, and account discovery has not been completed for the wallet, and sync is ongoing. It should remain invisible until the `updateSelectedAccount` function is called (including when the wallet account selection changes), and that function should determine the wallet state as explained above and only then make this view visible.